src/Lumi/Components2/Image.purs

```diff
@@ -0,0 +1,98 @@
+import Data.String as String
+import Data.Tuple.Nested ((/\))
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, PropsModifier, lumiComponent, ($$$))
+import Lumi.Components.Loader (loader)
+import Lumi.Components.Spacing (Space)
+import Lumi.Components.Svg (userSvg)
+import Lumi.Components2.Box as Box
+import Lumi.Styles (style, toCSS)
+import Lumi.Styles.Box (FlexAlign(..))
+import Lumi.Styles.Box as Styles.Box
+import Lumi.Styles.Image as Styles.Image
+import Lumi.Styles.Slat hiding (_interactive,slat) as Styles.Slat
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Events (handler_)
+import React.Basic.Hooks (JSX)
+import React.Basic.Hooks as Hooks
+
+type ImageProps
+  = ( src :: String
+    , placeholder :: Maybe JSX
+    )
+
+imageThumb :: LumiComponent ImageProps
+imageThumb =
+  unsafePerformEffect do
+    lumiComponent "Image" { src: "", placeholder: Nothing } \props -> Hooks.do
+      theme <- useTheme
+      loaded /\ setLoaded <- Hooks.useState' false
+      pure
+        $ E.element R.div'
+            { children:
+                [ if String.null props.src
+                    -- @TODO generate a placeholder svg
+                    then fromMaybe userSvg props.placeholder
+                    else
+                      Box.column
+                      $ Styles.Box._flex
+                      $ Styles.Box._align Center
+                      $ Styles.Box._justify Center
+                      $$$
+                        [ Monoid.guard (not loaded)
+                            $ loader { style: R.css { width: "20px", height: "20px", borderWidth: "2px" }, testId: Nullable.toNullable Nothing }
+                        , E.element R.img'
+                            { src: props.src
+                            , className: ""
+                            , css: E.css { maxWidth: E.percent 100.0 }
+                            , onLoad: handler_ $ setLoaded true
+                            }
+                        ]
+                ]
+            , className: props.className
+            , css: theme # toCSS Styles.Image.imageThumb <> props.css
+            }
+
+-- NOTE do we want to use the Space data type? (adheres to our 8px grid system..)
+-- or should we give consumers more flexibility?
+_customSize :: Space -> PropsModifier ImageProps
+_customSize size =
+  style \(LumiTheme theme) ->
+    E.css
+      { width: E.prop size
+      , height: E.prop size
+      }
+
+round :: LumiComponent ImageProps
+round = imageThumb <<< Styles.Image._round
+
+small :: LumiComponent ImageProps
+small = imageThumb <<< Styles.Image._small
+
+medium :: LumiComponent ImageProps
+medium = imageThumb <<< Styles.Image._medium
+
+large :: LumiComponent ImageProps
+large = imageThumb <<< Styles.Image._large
+
+extraLarge :: LumiComponent ImageProps
+extraLarge = imageThumb <<< Styles.Image._extraLarge
```

Comment:
I'm not sure I understand these values. `extraLarge` is a component here with `_extraLarge` preapplied. Isn't it simpler to just use `imageThumb` and `extraLarge` separately? Like
```purescript
imageThumb
$ extraLarge
$ ...
```

This brings me to another point, which is that I don't think the pattern of having style modifiers in `Lumi.Styles.*` wins us anything besides having to always import two modules instead of one when using a component.

Reply:
I now understand that it's supposed to be imported qualified and used as
```purescript
Image.large
$ ...
```

But still we'll have both `large` and `_large`, which is a potential source of confusion I think.

Reply:
Agreed. We never documented this but after Text2 Arthur and I briefly discussed doing away with the extra Styles module, and just defining those styles here. Splitting them doesn't gain us much and it becomes more difficult to use, so I'd lean toward doing the same here.

Reply:
I'd like to hear your (@kimispencer) and also @spicydonuts' thoughts on this matter too.

Reply:
Ah ok, yeah I was following the pattern I saw in `Box`, etc and thought it could give consumers more flexibility. But to your points, splitting it out doesn't give us much and just adds more cruft. I'll follow this new pattern and do away with the extra Styles module.

Reply:
I'm going to create an epic for lumi-components cleanup where we can document this a bit